client/src/components/features/Posts/Posts.js

```diff
@@ -3,27 +3,40 @@ import { PropTypes } from 'prop-types';
 import PostsList from '../PostsList/PostsList';
 import Spinner from '../../common/Spinner/Spinner';
 import Alert from '../../common/Alert/Alert';
+import Pagination from '../../common/Pagination/Pagination';
 
 class Posts extends React.Component {
   componentDidMount() {
-    const { loadPosts } = this.props;
-    loadPosts();
+    const { loadPostsByPage, initialPage, postsPerPage } = this.props;
+    loadPostsByPage(initialPage, postsPerPage);
   }
+
+  loadPostsPage = page => {
+    const { loadPostsByPage, postsPerPage } = this.props;
+    loadPostsByPage(page, postsPerPage);
+  };
   render() {
-    const { posts, request } = this.props;
+    const { posts, request, pages, pagination } = this.props;
+    const { loadPostsPage } = this;
 
     return (
       <div>
         {(request.pending || request.success === null) && <Spinner />}
         {!request.pending && request.error !== null && <Alert variant="error">Error: {request.error}</Alert>}
         {!request.pending && request.success && posts.length === 0 && <Alert variant="info">No posts</Alert>}
-        {!request.pending && request.success && posts.length > 0 && <PostsList posts={posts} />}
+        <PostsList posts={posts} />
+        {pagination && <Pagination pages={pages} onPageChange={loadPostsPage} />}
```

Comment:
Te dwa komponenty powinny się pokazać dopiero po spełnieniu `!request.pending && request.success && posts.length > 0` , dlaczego zdjąłeś ten warunek? W efekcie na wolnym połączeniu mam przy spinnerze `>` u dołu strony

Reply:
Zdjąłem warunek - ponieważ blokował zmianę kropek. Po prostu z tym warunkiem nie działała zmiana kropek, przeskakiwały strony, a nie zmieniały się kropki. Stwierdziłem, że PostList jest niewidoczny w sytuacji, gdy postów nie ma i zlikwidowałem ten warunek. Jeśli chodzi o > u dołu strony, to nie zauważyłem tego błędu. Przyjrzę się temu fragmentowi.